src/RA_MemManager.h

```diff
@@ -29,31 +19,44 @@ class MemManager
     class BankData
     {
     public:
-        BankData()
-            : Reader(nullptr), Writer(nullptr), BankSize(0)
+        // This should give NRVO ("named return value optimization"), since not
+        // all constructors are defaulted you have to mark them constexpr
+        // explicitly
+        inline constexpr BankData() noexcept = default;
+        ~BankData() noexcept = default;
+
+        inline constexpr BankData(_RAMByteReadFn* pReadFn, _RAMByteWriteFn* pWriteFn, size_t nBankSize) noexcept :
+            Reader{ pReadFn },
+            Writer{ pWriteFn },
+            BankSize{ nBankSize }
         {
         }
+        
+        // If you are trying to use copy epsilon technique we should delete
+        // moving as well but we may need it for unique_ptr later. Deleting
+        // copying should give it sufficient optimization.
 
-        BankData(_RAMByteReadFn* pReadFn, _RAMByteWriteFn* pWriteFn, size_t nBankSize)
-            : Reader(pReadFn), Writer(pWriteFn), BankSize(nBankSize)
-        {
-        }
-
-    private:
-        //	Copying disabled
-        BankData(const BankData&);
-        BankData& operator=(BankData&);
+        //	Copying deleted
+        BankData(const BankData&) = delete;
+        BankData& operator=(BankData&) = delete;
+        inline constexpr BankData(BankData&&) noexcept = default;
+        inline constexpr BankData& operator=(BankData&&) noexcept = default;
 
     public:
-        _RAMByteReadFn * Reader;
-        _RAMByteWriteFn* Writer;
-        size_t BankSize;
+        _RAMByteReadFn* Reader{ nullptr };
+        _RAMByteWriteFn* Writer{ nullptr };
+        size_t BankSize{ std::size_t{} };
     };
 
 public:
-    MemManager();
-    virtual ~MemManager();
-
+    // Default constructors aren't allowed to throw but the default does so we have to define it
+    MemManager() noexcept; 
+
+    MemManager(const MemManager&) = delete;
+    MemManager& operator=(MemManager&) = delete;
+    MemManager(MemManager&&) = delete;  // might throw but there's a global so moving is unnecessary for now
+    MemManager& operator=(MemManager&&) noexcept = delete;
+    ~MemManager() noexcept;
```

Comment:
Virtual destructors or for inheritance only for polymorphic behavior.
I didn't see any classes derived from MemManager.